custom/opm/utils.py

```diff
@@ -28,6 +29,7 @@ def _safeint(value):
         return 0
 
 
+@memoized
```

Comment:
How does this work?  I haven't seen the memoized decorator applied to a class definition before.  I think what we want in this case is for the class to be instantiated lazily on demand, and thereafter referred to as a singleton.  This class shouldn't grow over time, so it's probably fine to keep it around in memory, although a timeout approach such as quickcache (which also can do in-memory caching) would work too.

I'm not familiar with memoizing a class definition, but if it acts like it sounds, that's probably fine.